File: Docker/Other/springboot-maven-mysql/src/main/java/com/example/employees/controller/HealthController.java
package com.example.employees.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import java.time.Instant;
import java.util.Map;

/**
 * Health check endpoint.
 * Used by Docker healthcheck, load balancers, and monitoring tools.
 */
@RestController
@RequiredArgsConstructor
public class HealthController {

    private final JdbcTemplate jdbcTemplate;

    @GetMapping("/health")
    public ResponseEntity<Map<String, Object>> health() {
        try {
            jdbcTemplate.queryForObject("SELECT 1", Integer.class);
            return ResponseEntity.ok(Map.of(
                "status", "ok",
                "database", "ok",
                "timestamp", Instant.now().toString()
            ));
        } catch (Exception e) {
            return ResponseEntity.status(500).body(Map.of(
                "status", "error",
                "database", "unreachable",
                "detail", e.getMessage()
            ));
        }
    }
}

File: Docker/Other/laravel-mysql/app/Http/Controllers/HealthController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\JsonResponse;
use Illuminate\Support\Facades\DB;

class HealthController extends Controller
{
    public function check(): JsonResponse
    {
        try {
            DB::select('SELECT 1');
            return response()->json([
                'status'    => 'ok',
                'database'  => 'ok',
                'timestamp' => now()->toIso8601String(),
            ]);
        } catch (\Exception $e) {
            return response()->json([
                'status'   => 'error',
                'database' => 'unreachable',
                'detail'   => $e->getMessage(),
            ], 500);
        }
    }
}
